students/Mintae_K/lesson03/mailroom.py

```diff
@@ -0,0 +1,60 @@
+# Sending a Thank you
+def sendingthank(donors):
+    f_name = input('Please type the Full Name or list> ')
+    while f_name == 'list':
+        print(' ')
+        for x in donors:
+            print(x[0])
+        print(' ')
+        f_name = input('Please type the Full Name or list> ')
+    for x in range(len(donors)):
+        if f_name == donors[x][0]:
+            name_exists = True
+            break
+        else:
+            name_exists = False
+    if name_exists is False:
+        s = 'New donor, please type the donation amount from {}> '
+        amount = input(s.format(f_name))
+        donors = donors + [[f_name, float(amount)]]
+    else:
+        s = 'Existing donor, please type the donation amount from {}> '
+        amount = input(s.format(f_name))
+        donors[x] = donors[x] + [float(amount)]
+    print('\n', "Thank you {} for the generous donation!".format(f_name), '\n')
+    return donors
+
+
+# Creating Report
+def createreport(donors):
+    s = '\n, {:<20}| {:<10} | {:<10}| {:<20}'
+    print(s.format('Donor Name', 'Total Given', "Num Gift", "Average Gift"))
+    print('-' * 60)
+    for x in range(len(donors)):
+        total = sum(donors[x][1:])
+        num_gift = len(donors[x]) - 1
+        s = '{:<20}  $ {:<10.2f}  {:^12} $ {:<20.2f}'
+        print(s.format(donors[x][0], total, num_gift, total / num_gift))
```

Comment:
This report should be sorted (check assignment requirements), please add sorting in the next revision.